### ISSUE TITLE
Empty grain size is quite big

### ISSUE THREAD

**Issue by zertrin** [PDIS/codimd-sandstorm#12]
Hello!

I just tried CodiMD on sandstorm and quite like the app.

However, a newly created grain with no content already weigh 48.4 MB!
This overhead for just a text note seems very excessive to me.

Do you think there is anything that can be done to make the grains more lightweight in the future?

**Comment by nicolas-arnold** [comment]
I made the same observation. There also exists SandMD (Last updated: 2018 Dec 27) whereas CodiMD seems to be more up to date (Last updated: 2020 Apr 29).

I tried CodiMD as I had several issues with unresponsive SandMD pads, but an empty pad over there has about 90kb for me. As both have the same origin, I consider ~50 MB for an empty pad indeed a bug and it's a reason not to use this version, although it might be more stable in general.

Has that ever been so and is there an explanation for why it's so big? What is stored individually besides the text and some metadata itself upfront?

**Comment by bingluen** [comment]
According to [CodiMD official document](https://hackmd.io/c/codimd-documentation/%2Fs%2Fcodimd-db-connection):

> We will only support PostgreSQL and MySQL in future database migrations, if any.

We choose MySQL as database to storage data.
SandMD, we previously published, use SQLite file to storage data.
That's why CodiMD's grain size is bigger than SandMD.

I refer [platform stacks](https://docs.sandstorm.io/en/latest/vagrant-spk/platform-stacks/) from Sandstorm official document to build MySQL environment for CodIMD. 

It maybe not best practice that I current implement on CodiMD's Sandstorm app.

Welcome to discuss, if you have any idea to optimize

**Comment by zertrin** [comment]
This is very unfortunate, and in my case, a deal-breaker for usage of CodiMD on sandstorm. 

A single overhead of 48 MB for all notes would be acceptable (e.g. if I were to install CodiMD directly, not on sandstorm), but a repeated overhead of 48 MB per note is just not acceptable for me.

Thanks for the explanation though! Good luck in finding a solution 🙂

**Comment by ocdtrekkie** [comment]
Yeah, unfortunately, MySQL isn't a *great* fit for Sandstorm apps because of it's overhead. It isn't too big a deal with TinyTinyRSS, for instance, where you're likely to only have one or two grains, but it can be painful for single-document apps.

Presumably someone could update SandMD, but I assume @bingluen is only going to invest time/effort in supporting the package that PDIS is using.
